src/grids.jl

```diff
@@ -212,6 +212,15 @@ get_colat(::Type{<:FullClenshawGrid},nlat_half::Integer) = [j/(2nlat_half)*π fo
 get_quadrature_weights(::Type{<:FullClenshawGrid},nlat_half::Integer) = clenshaw_curtis_weights(nlat_half)
 full_grid(::Type{<:FullClenshawGrid}) = FullClenshawGrid    # the full grid with same latitudes
 
+
+# GPU methods FullClenshawGrid
+
+function Adapt.adapt_structure(to, fcg::FullClenshawGrid)
+    FullClenshawGrid(Adapt.adapt(to, fcg.data),
+                     fcg.nlat_half)
+end
```

Comment:
I don't know whether there's anything special within the `Adapt` framework that wouldn't allow this, but one should be able to dispatch that with a single function
```julia
function Adapt.adapt_structure(to, grid::Grid) where {Grid<:AbstractGrid}
    Grid(Adapt.adapt(to, grid.data), grid.nlat_half)
end
```
all `Grid<:AbstractGrid` now use `nlat_half` as resolution parameter.

Reply:
So, do I understand this correctly: I would just put what you wrote here into the (new) Grid.jl file and then it should be dispatched to all other files in the Grids folder (e.g. full_grids.jl), since there, all the structs are for AbstractGrid types?

Reply:
Yes, this should work for all `Grid<:AbstractGrid` as we changed to a uniform interface: The grid points are always in `grid.data` stored as `::Vector` and the resolution parameter is always half the number of latitude rings `nlat_half`. So one function should be good for all grids, but test it, I don't know whether the Adapt framework requires anything special.

Reply:
Okay, thanks!

Reply:
I like Milan's suggestion. Could you change it like that, and then I'll also have a look into solving the merge conflicts that are now present due to the updated `gpu` branch